addons/mail/static/src/models/user_notification_manager.js

```diff
@@ -0,0 +1,149 @@
+            }
+        },
+        /**
+         * @private
+         * @returns {HTMLAudioElement}
+         */
+        _computeAudio() {
+            const audioElement = new Audio();
+            audioElement.src = audioElement.canPlayType("audio/ogg; codecs=vorbis")
+                ? url('/mail/static/src/audio/ting.ogg')
+                : url('mail/static/src/audio/ting.mp3');
+            return audioElement;
+        },
+        /**
+         * Determines whether or not sending native notification is
+         * allowed.
+         *
+         * @private
+         * @returns {boolean}
+         */
+        _computeCanSendNativeNotification() {
+            return (
+                this.messaging.browser.Notification &&
+                this.messaging.browser.Notification.permission === 'granted'
+            );
+        },
+        /**
+         * Method to be called when the users click on a notification.
+         *
+         * @private
+         */
+        _onClickNotification() {
+            window.focus();
+            this.notification.close();
+        },
+        /**
+         * Play the notification sound.
+         */
+        async _playAudio() {
+            if (this.canPlayAudio) {
+                try {
+                    await this.audio.play();
+                } catch {
+                    // Ignore errors due to the user not having interracted
+                    // with the page before playing the sound.
+                }
+            }
+        },
+        /**
+         * Send a native notification.
+         *
+         * @param {string} title
+         * @param {string} message
+         */
+        _sendNativeNotification(title, message) {
+            const notification = new Notification(
+                // The native Notification API works with plain text and not HTML
+                // unescaping is safe because done only at the **last** step
+                _.unescape(title),
+                {
+                    body: _.unescape(message),
+                    icon: this.icon,
+                }
+            );
+            notification.addEventListener('click', this._onClickNotification);
+            this.update({ notification });
+        },
+        /**
+         * Send a notification through the notification service.
+         *
+         * @param {string} message
+         * @param {Object} options
+         */
+        _sendOdooNotification(message, options) {
+            this.messaging.env.services['notification'].add(message, options);
+            if (this.messaging.env.services['multi_tab'].isOnMainTab()) {
+                this._playAudio();
+            }
+        },
+    },
+    fields: {
+        /**
+         * HTMLAudioElement used to play sound when a notification is
+         * sent.
+         */
+        audio: attr({
+            compute: '_computeAudio',
+        }),
+        canPlayAudio: attr({
+            default: typeof(Audio) != 'undefined',
```

Comment:
`!==`, also can't compare to undefined directly instead of using typeof?

Reply:
Can't, if a symbol is not defined a `ReferenceError` will be raised